### DISCUSSION TITLE
Flashing build to ESP32

### DISCUSSION THREAD

**Discussion by SixAxisSig** [StuckAtPrototype/Racer#45] [Q&A]
Racer Community,

I’ve just received my boards last week. I’m new to ESP32 and have been able to compile the "firmware" using ESP-IDF 5.3 PowerShell (getting some warnings, but completing the build). 

When attempting to flash the build to the ESP32, I am getting "Failed to connect to ESP32-H2: No serial data received." I have used two separate USB to Serial cables, confirmed the chip target to ESP32h2, and still the connection issue. I've checked supply voltages and confirmed the "BOOT" button does pull to GND when pressed. 

I'm lost here and I bet its something simple I've either configured wrong, or a step I'm missing. Please help if you have successfully flashed your builds. Any assistance is greatly appreciated. 

In return, I offer my design skillset. I’ve designed a body for my racer, the Bonk Truck Racer. If interested, I’ll share my STL files and I’ll take any suggestions for additional bodies. (working on a VW Bug body now)

Thanks for this excellent project and I’m looking forward to growing this community. 

![IMG_5074](https://github.com/user-attachments/assets/d6b8cda6-ad71-4884-a748-2627db7c3ecb)
![IMG_5072](https://github.com/user-attachments/assets/a9dc5b59-6a53-4a84-bdba-197a5ef9099f)

**Comment by Quas7** [comment]
First guess: The USB port is not connected to the esp32 in the current revision so you need to use a serial-to-usb converter for upload via the programming pins.
